src/Text/Pandoc/Readers/DocBook.hs

```diff
@@ -900,15 +905,35 @@ parseBlock (Elem e) =
                                                 Just ws' -> let tot = sum ws'
                                                             in  ColWidth . (/ tot) <$> ws'
                                                 Nothing  -> replicate numrows ColWidthDefault
-                      let toRow = Row nullAttr . map simpleCell
+                      let toRow :: [Cell] -> Row
+                          toRow = Row nullAttr
+                          toHeaderRow :: [Cell] -> [Row]
                           toHeaderRow l = if null l then [] else [toRow l]
                       return $ table (simpleCaption $ plain capt)
                                      (zip aligns widths)
                                      (TableHead nullAttr $ toHeaderRow headrows)
                                      [TableBody nullAttr 0 [] $ map toRow bodyrows]
                                      (TableFoot nullAttr [])
          isEntry x  = named "entry" x || named "td" x || named "th" x
-         parseRow = mapM (parseMixed plain . elContent) . filterChildren isEntry
+         parseEntry :: PandocMonad m => [String] -> Element -> DB m Cell
+         parseEntry cn el = do
+           let colDistance :: String -> String -> ColSpan
+               colDistance sa ea = do
+                 let si = elemIndex sa cn
+                 let ei = elemIndex ea cn
+                 if (isJust si) && (isJust ei)
+                   then ColSpan $ (fromJust ei) - (fromJust si) + 1
+                   else 1
+           let toColSpan :: Element -> ColSpan
+               toColSpan en = do
+                 let sa = findAttr (unqual "namest") en
+                 let ea = findAttr (unqual "nameend") en
+                 if (isJust sa) && (isJust ea)
+                   then colDistance (fromJust sa) (fromJust ea)
+                   else 1
```

Comment:
you might want to rewrite this to something like:

```haskell
let mStrt = findAttr (unqual "namest") en
let mEnd = findAttr (unqual "nameend") en
case (mStrt, mEnd) of
  (Just start, Just end) -> colDistance start end
  _ -> 1
```

Reply:
done - thanks - much cleaner!